tests/planner/parabolic/test_ParabolicSmoother.cpp

```diff
@@ -0,0 +1,231 @@
+
+    mNonStraightLineLength = initNonStraightLine();
+  }
+
+  double initNonStraightLine()
+  {
+    mNonStraightLine = std::make_shared<Interpolated>
+            (mStateSpace, mInterpolator);
+
+    auto state = mStateSpace->createState();
+    Vector2d p1(1., 1.), p2(2., 1.2), p3(2.2, 2.),
+            p4(3., 2.2), p5(3.2, 3.), p6(4.,3.2);
+    state.setValue(p1);
+    mNonStraightLine->addWaypoint(0., state);
+    state.setValue(p2);
+    mNonStraightLine->addWaypoint(1., state);
+    state.setValue(p3);
+    mNonStraightLine->addWaypoint(2., state);
+    state.setValue(p4);
+    mNonStraightLine->addWaypoint(3., state);
+    state.setValue(p5);
+    mNonStraightLine->addWaypoint(4., state);
+    state.setValue(p6);
+    mNonStraightLine->addWaypoint(5, state);
+
+    double length = (p2-p1).norm() + (p3-p2).norm()
+            + (p4-p3).norm() + (p5-p4).norm()
+            + (p6-p5).norm();
+    return length;
+  }
+
+  double getLength(aikido::trajectory::Spline* spline)
+  {
+    double length = 0.0;
+    auto currState = mStateSpace->createState();
+    auto nextState = mStateSpace->createState();
+    Eigen::VectorXd currVec, nextVec;
+    for(size_t i=0; i<spline->getNumWaypoints()-1;++i)
+    {
+      spline->getWaypoint(i, currState);
+      spline->getWaypoint(i+1, nextState);
+      mStateSpace->logMap(currState, currVec);
+      mStateSpace->logMap(nextState, nextVec);
+      length += (nextVec-currVec).norm();
+    }
+    return length;
+  }
+
+  aikido::util::RNGWrapper<std::mt19937> mRng;
+  std::shared_ptr<R2> mStateSpace;
+  Eigen::Vector2d mMaxVelocity;
+  Eigen::Vector2d mMaxAcceleration;
+
+  std::shared_ptr<GeodesicInterpolator> mInterpolator;
+  std::shared_ptr<Interpolated> mStraightLine;
+  std::shared_ptr<Interpolated> mNonStraightLine;
+
+  double mNonStraightLineLength;
+  double mTimelimit = 60.0;
+  const double mTolerance = 1e-5;
+};
+
+TEST_F(ParabolicSmootherTests, convertInterpolatedToSpline)
+{
+    auto spline = convertToSpline(*mStraightLine.get());
+
+    auto splineState = mStateSpace->createState();
+    auto interpolatedState = mStateSpace->createState();
+    Eigen::VectorXd splineVec, interpolatedVec;
+    Eigen::VectorXd splineTangent, interpolatedTangent;
+
+    const double stepSize = 1e-3;
+    aikido::util::StepSequence seq(stepSize, true,
+                                   spline->getStartTime(),
+                                   spline->getEndTime());
+    for(double t : seq)
+    {
+        spline->evaluate(t, splineState);
+        mStraightLine->evaluate(t, interpolatedState);
+        mStateSpace->logMap(splineState, splineVec);
+        mStateSpace->logMap(interpolatedState, interpolatedVec);
+        EXPECT_EIGEN_EQUAL(splineVec, interpolatedVec, mTolerance);
+
+        spline->evaluateDerivative(t, 1, splineTangent);
+        mStraightLine->evaluateDerivative(t, 1, interpolatedTangent);
+        EXPECT_EIGEN_EQUAL(splineTangent, interpolatedTangent, mTolerance);
+    }
+}
```

Comment:
Add tests for:
- More than one segment, e.g. the non-striaghtline trajectory.
- A trajectory with a non-zero start time.